crates/network-primitives/src/traits.rs

```diff
@@ -51,6 +51,14 @@ pub trait ReceiptResponse {
 
     /// EIP-7702 Authorization list.
     fn authorization_list(&self) -> Option<&[SignedAuthorization]>;
+
+    /// Returns the cumulative gas used at this receipt.
+    fn cumulative_gas_used(&self) -> u128;
+
+    /// The post-transaction state root (pre Byzantium)
+    ///
+    /// EIP98 makes this field optional.
+    fn state_root(&self) -> Option<&B256>;
```

Comment:
```suggestion
    fn state_root(&self) -> Option<B256>;
```

Reply:
this is at odds with the other trait functions. I'd like to unify this and return a copy